### PR TITLE
[JENKINS-45349] - Restore Remoting Test Client codebase from Maven Repo

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I was unable to find location of this repo, hence I have recovered sources from Maven repo and added a new POM.xml for it

https://issues.jenkins-ci.org/browse/JENKINS-45349

Once the change is integrated, I will add fixes for JENKINS-45349 in a separate pull request

@reviewbybees @jtnord

### CODE REVIEW

File: src/it/remoting-test-client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  <properties>
+    <java.level>7</java.level>
+    <build.type>private</build.type>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.jenkins-ci.main</groupId>
+      <artifactId>remoting</artifactId>
+      <version>2.30</version>
```
Comment (oleg-nenashev) [inline src/it/remoting-test-client/pom.xml]: I have just taken something random what has no `checkRoles` in callable

**Comment by jtnord** [comment]
source is here: https://github.com/jenkinsci/remoting/tree/remoting-test-client-1.0

**Comment by oleg-nenashev** [comment]
Seems GitHub search didn't pick this branch due to the whatever reason

**Comment by oleg-nenashev** [comment]
I will close this PR and move the original code

**Comment by oleg-nenashev** [comment]
Closing in favor of #181

**Pull request closed without merging**